src/util/setUnreadCount.js

```diff
@@ -0,0 +1,31 @@
+import db from "../options/containers/notifications/index-pouch.js"
+import PouchDB from "pouchdb"
+
+
+export default function setUnreadCount(itemToCount) {
```

Comment:
All this asynchronous/promise based code can be cleaned up by using `asyncawait`. Essentially allows you to wait on the resolution of the Promise, using `await`, rather than passing a callback function in `.then()`, and errors can be handled using `try ... catch` rather than a callback in `.catch()`. The general expected flow can go in the `try {}` block, and in any case of an error being encountered in the async logic (the `PouchDB.allDocs` call, in this case), it will jump to the `catch {}` block. So consequential logic doesn't need to be nested in different functions. 

Have a read up on it or have a look at some of the other parts in the codebase (most of our code is using it). As an example, this function could be rewritten like:
```js
export default async function setUnreadCount(itemToCount) {
    try {
        const response = await db.allDocs({
            include_docs: true,
            attachments: true,
            startkey: "notif",
            endkey: "notif\ufff0",
        })

        response.rows.forEach(element => {
            if (!element.doc.viewed) {
                itemToCount++
            }
        }
        return itemToCount
    } catch (err) {
        console.log("err", err)
    }
}
```